applications/crossbar/src/crossbar_init.erl

```diff
@@ -63,10 +63,54 @@ start_link() ->
 %% @public
 %% @doc Load a crossbar module's bindings into the bindings server
 %%--------------------------------------------------------------------
--spec start_mod(atom() | string() | binary()) -> any().
-start_mod(CBMod) when not is_atom(CBMod) -> start_mod(kz_util:to_atom(CBMod, 'true'));
-start_mod(CBMod) -> CBMod:init().
 
+-spec is_versioned_module(binary()) -> boolean().
+is_versioned_module(Module) ->
+    Mod = lists:reverse(binary_to_list(Module)),
+    case Mod of
+        "1v_" ++ _ -> 'true';
+        "2v_" ++ _ -> 'true';
+        _ -> false
+    end.
+
+-spec start_mod(atom() | string() | binary()) -> 'ok' | {'error', any()}.
+start_mod(CBMod) when not is_atom(CBMod) ->
```

Comment:
Since is_versioned_module expects binaries (or it will crash), then the `not is_atom` guard should actually be `is_binary`